src/internal/packager2/load.go

```diff
@@ -0,0 +1,212 @@
+			if err != nil {
+				return err
+			}
+		}
+		dst, err := os.Create(filepath.Join(packageDir, header.Name))
+		if err != nil {
+			return err
+		}
+		defer dst.Close()
+		_, err = io.Copy(dst, f)
+		if err != nil {
+			return err
+		}
+		pathsExtracted = append(pathsExtracted, header.Name)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Load the package paths
+	pkgPaths := layout.New(packageDir)
+	pkgPaths.SetFromPaths(pathsExtracted)
+	pkg, _, err := pkgPaths.ReadZarfYAML()
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Filter is not persistently applied.
+	pkg.Components, err = opt.Filter.Apply(pkg)
+	if err != nil {
+		return nil, err
+	}
+	if err := pkgPaths.MigrateLegacy(); err != nil {
+		return nil, err
+	}
+	if !pkgPaths.IsLegacyLayout() {
+		if err := sources.ValidatePackageIntegrity(pkgPaths, pkg.Metadata.AggregateChecksum, isPartial); err != nil {
+			return nil, err
+		}
+		if opt.SkipSignatureValidation {
+			if err := sources.ValidatePackageSignature(ctx, pkgPaths, opt.PublicKeyPath); err != nil {
+				return nil, err
+			}
+		}
+	}
+	for _, component := range pkg.Components {
+		if err := pkgPaths.Components.Unarchive(component); err != nil {
+			if errors.Is(err, layout.ErrNotLoaded) {
+				_, err := pkgPaths.Components.Create(component)
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				return nil, err
+			}
+		}
+	}
+	if pkgPaths.SBOMs.Path != "" {
+		if err := pkgPaths.SBOMs.Unarchive(); err != nil {
+			return nil, err
+		}
+	}
+	return pkgPaths, nil
+}
+
+func identifySource(src string) (string, error) {
+	parsed, err := url.Parse(src)
+	if err == nil && parsed.Scheme != "" && parsed.Host != "" {
+		return parsed.Scheme, nil
+	}
+	if strings.HasSuffix(src, ".tar.zst") || strings.HasSuffix(src, ".tar") {
+		return "tarball", nil
+	}
+	if strings.Contains(src, ".part000") {
+		return "split", nil
+	}
+	return "", fmt.Errorf("unknown source %s", src)
+}
+
+func assembleSplitTar(src, tarPath string) error {
+	pattern := strings.Replace(src, ".part000", ".part*", 1)
+	splitFiles, err := filepath.Glob(pattern)
+	if err != nil {
+		return fmt.Errorf("unable to find split tarball files: %w", err)
+	}
+	// Ensure the files are in order so they are appended in the correct order
+	sort.Strings(splitFiles)
+
+	tarFile, err := os.Create(tarPath)
+	if err != nil {
+		return err
+	}
+	defer tarFile.Close()
+
+	var pkgData types.ZarfSplitPackageData
+	for _, splitFile := range splitFiles {
+		f, err := os.Open(splitFile)
+		if err != nil {
+			return fmt.Errorf("unable to open file %s: %w", splitFile, err)
+		}
+		defer f.Close()
+		_, err = io.Copy(tarFile, f)
+		if err != nil {
+			return fmt.Errorf("unable to copy file %s: %w", splitFile, err)
+		}
+		err = f.Close()
+		if err != nil {
+			return fmt.Errorf("unable to close file %s: %w", splitFile, err)
+		}
+	}
+	if err := helpers.SHAsMatch(tarPath, pkgData.Sha256Sum); err != nil {
```

Comment:
How does `pkgData.Sha245Sum` get set in the split tar  - I'm not seeing where we are unmarshaling the package metadata anymore

Reply:
The shasum that exists in the package is the same as the tar package that is assembled at the end. In the current implementation we are not actually checking the shasum of the assembled tar, only what the metadata says. This means that the resulting tar package could be something completely different. Shasum checking is done after the tar is assembled.

This comment did however made me realize that like the pull command we dont have a flag to set the shasum for mirror. So I have added that.